packages/data-provider/src/utils/dictionaryUtils.ts

```diff
@@ -19,9 +22,9 @@ export const getSchemaFieldNames = (dictionary: SchemasDictionary, entityType: s
 			field.restrictions && 'required' in field.restrictions ? field.restrictions.required : undefined;
 
 		if (requiredRestriction) {
-			fieldNamesMapped.required.push(field.name);
+			fieldNamesMapped.required.push(field.meta?.displayName.toString() || field.name);
```

Comment:
is displayName a guaranteed value, when meta exists? otherwise, further conditional chaining could be useful here

```suggestion
			fieldNamesMapped.required.push(field.meta?.displayName?.toString() || field.name);
```